rclcpp/include/rclcpp/create_client.hpp

```diff
@@ -63,12 +64,40 @@ create_client(
   std::shared_ptr<node_interfaces::NodeBaseInterface> node_base,
   std::shared_ptr<node_interfaces::NodeGraphInterface> node_graph,
   std::shared_ptr<node_interfaces::NodeServicesInterface> node_services,
+  std::shared_ptr<node_interfaces::NodeClockInterface> node_clock,
+  const std::string & service_name,
+  const rmw_qos_profile_t & qos_profile,
+  rclcpp::CallbackGroup::SharedPtr group,
+  bool enable_service_introspection)
```

Comment:
While we're adding/modifying API, it might be a good time to consider introducing an options struct, analogous to those for publishers and [subscriptions](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/subscription_options.hpp#L39). This is where we could bundle together the optional parameters: clock interface, callback group, and service introspection flag.

Doing this now would avoid potential API breaks (or deprecation cycles) in the future. I could look at opening a separate PR for this and then we could build on top of it here (depending how much work it turns out to be).

Related issue: https://github.com/ros2/rclcpp/issues/1989

Reply:
Yeah, I agree that it would be nice but a little outside of scope for this PR, though. We can open up a follow-up PR to introduce a ServiceOptions struct later on